Link both global-nest fix files and non-nest ones at the same time

guoqing-noaa opened the pull request:
# Description
  This PR enables linking both global-nest fix files and non-nest ones at the same time and users can run both nesting and non-nesting experiments at the same time without worries about what fix files to be linked. 

  Resolves #2631 

# Type of change
- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
- Clone and build on Hera/Jet
- Forecast-only on Hera/Jet

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

guoqing-noaa committed:
Link both global-nest fix files and non-nest ones at the same time

guoqing-noaa committed:
double quoting even when variables don't contain special characters

aerorahul commented on sorc/link_workflow.sh:
```diff
@@ -10,7 +10,7 @@ function usage() {
   scripts in sequence.
 
-Usage: ${BASH_SOURCE[0]} [-h][-o][--nest]
+Usage: ${BASH_SOURCE[0]} [-h][-o]
```
@guoqing-noaa 
I don't think linking 2 sets of fix files is a good idea.
We should target consolidation rather than duplication.

guoqing-noaa replied:
Thanks for the comment, @aerorahul !  I am open to a suitable solution.

Our experiences showed that it is troublesome that we need to relink fix files when one wants to transition from running nesting experiments to running a regular non-nest experiment. Most times we forgot to do that and got crashes. And we cannot relink those fix files in the process of a launched experiment and have to wait for its completion.
Also, this means we cannot run nesting experiments and non-nesting experiments at the same time.

My current solution is to link all directories in fix to fix/nest but update fix/nest/orog and fix/nest/ugwd. 
To my understanding,  this does not cause any harm to the current global workflow. Let me know if you find any problems with this. Thanks!

guoqing-noaa replied:
@aerorahul Do you mean we keep the "--nest" option so that the solution only applies to those who would run nesting experiments?

guoqing-noaa committed:
Merge branch 'develop' into pr_nest_fix_files

guoqing-noaa commented:
@aerorahul and @WalterKolczynski-NOAA Thanks a lot for the discussion in today's meeting!  
I found that there are many more files involved regarding changing ${FIXgfs}/orog to ${FIXorog}. 11 files (from scripts/, parm/, ush/) will be affected:
```
scripts/exgdas_enkf_sfc.sh               
scripts/exglobal_atmos_sfcanl.sh         
scripts/exglobal_forecast.sh             
parm/config/gfs/config.base              
parm/ufs/fix/gfs/atmos.fixed_files.yaml  
parm/ufs/fix/gfs/land.fixed_files.yaml   
parm/ufs/fix/gfs/ocean.fixed_files.yaml  
ush/forecast_predet.sh                   
ush/gaussian_sfcanl.sh                   
ush/global_cycle.sh                      
ush/global_cycle_driver.sh 
```
Where `ush/global_cycle.sh` and `ush/global_cycle_driver.sh` are linked from another repository `ufs_utils`. 
I am hesitant to make changes to so many files. However, I can update them if you agree that this is the goal we should go. 

FYI, here are the output from `grep '\/orog' -ir scripts/* parm/* ush/*`:
```
scripts/exgdas_enkf_sfc.sh:            ${NLN} "${FIXgfs}/orog/${CASE}/${CASE}_grid.tile${n}.nc"     "${DATA}/fngrid.${cmem}"                                           
scripts/exgdas_enkf_sfc.sh:            ${NLN} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile${n}.nc" "${DATA}/fnorog.${cmem}"                               
scripts/exgdas_enkf_sfc.sh:            ${NLN} "${FIXgfs}/orog/${CASE}/${CASE}_grid.tile${n}.nc"      "${DATA}/fngrid.${cmem}"                                          
scripts/exgdas_enkf_sfc.sh:            ${NLN} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile${n}.nc" "${DATA}/fnorog.${cmem}"                               
scripts/exglobal_atmos_sfcanl.sh:        ${NLN} "${FIXgfs}/orog/${CASE}/${CASE}_grid.tile${n}.nc"                         "${DATA}/fngrid.00${n}"                      
scripts/exglobal_atmos_sfcanl.sh:        ${NLN} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile${n}.nc"                     "${DATA}/fnorog.00${n}"          
scripts/exglobal_atmos_sfcanl.sh:    ${NLN} "${FIXgfs}/orog/${CASE}/${CASE}_grid.tile${n}.nc"                       "${DATA}/fngrid.00${n}"                            
scripts/exglobal_atmos_sfcanl.sh:    ${NLN} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile${n}.nc"                   "${DATA}/fnorog.00${n}"                
scripts/exglobal_forecast.sh:##         1. computing grid, ${FIXgfs}/orog/$CASE/${CASE}_grid.tile${n}.nc                                                               
scripts/exglobal_forecast.sh:##         2. orography data, ${FIXgfs}/orog/$CASE/${CASE}.mx${OCNRES}_oro_data.tile${n}.nc                                               
scripts/exglobal_forecast.sh:##         3. mosaic data, ${FIXgfs}/orog/$CASE/${CASE}_mosaic.nc                                                                         
parm/config/gfs/config.base:export FIXorog="${FIXgfs}/orog"                                                                                                            
parm/ufs/fix/gfs/atmos.fixed_files.yaml:  - [$(FIXgfs)/orog/$(CASE)/$(CASE)_mosaic.nc, $(DATA)/INPUT/grid_spec.nc]                                                     
parm/ufs/fix/gfs/atmos.fixed_files.yaml:  - [$(FIXgfs)/orog/$(CASE)/$(CASE)_grid.tile1.nc, $(DATA)/INPUT/]                                                             
parm/ufs/fix/gfs/atmos.fixed_files.yaml:  - [$(FIXgfs)/orog/$(CASE)/$(CASE)_grid.tile2.nc, $(DATA)/INPUT/]                                                             
parm/ufs/fix/gfs/atmos.fixed_files.yaml:  - [$(FIXgfs)/orog/$(CASE)/$(CASE)_grid.tile3.nc, $(DATA)/INPUT/]                                                             
parm/ufs/fix/gfs/atmos.fixed_files.yaml:  - [$(FIXgfs)/orog/$(CASE)/$(CASE)_grid.tile4.nc, $(DATA)/INPUT/]                                                             
parm/ufs/fix/gfs/atmos.fixed_files.yaml:  - [$(FIXgfs)/orog/$(CASE)/$(CASE)_grid.tile5.nc, $(DATA)/INPUT/]                                                             
parm/ufs/fix/gfs/atmos.fixed_files.yaml:  - [$(FIXgfs)/orog/$(CASE)/$(CASE)_grid.tile6.nc, $(DATA)/INPUT/]                                                             
parm/ufs/fix/gfs/land.fixed_files.yaml:     # Files from FIXgfs/orog/C??/sfc                                                                                           
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).facsf.tile1.nc, $(DATA)/]                                                
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).facsf.tile2.nc, $(DATA)/]                                                
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).facsf.tile3.nc, $(DATA)/]                                                
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).facsf.tile4.nc, $(DATA)/]                                                
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).facsf.tile5.nc, $(DATA)/]                                                
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).facsf.tile6.nc, $(DATA)/]                                                
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).maximum_snow_albedo.tile1.nc, $(DATA)/]                                  
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).maximum_snow_albedo.tile2.nc, $(DATA)/]                                  
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).maximum_snow_albedo.tile3.nc, $(DATA)/]                                  
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).maximum_snow_albedo.tile4.nc, $(DATA)/]                                  
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).maximum_snow_albedo.tile5.nc, $(DATA)/]                                  
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).maximum_snow_albedo.tile6.nc, $(DATA)/]                                  
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).slope_type.tile1.nc, $(DATA)/]                                           
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).slope_type.tile2.nc, $(DATA)/]                                           
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).slope_type.tile3.nc, $(DATA)/]                                           
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).slope_type.tile4.nc, $(DATA)/]                                           
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).slope_type.tile5.nc, $(DATA)/]                                           
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).slope_type.tile6.nc, $(DATA)/]                                           
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).snowfree_albedo.tile1.nc, $(DATA)/]                                      
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).snowfree_albedo.tile2.nc, $(DATA)/]                                      
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).snowfree_albedo.tile3.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).maximum_snow_albedo.tile4.nc, $(DATA)/]                         [50/4604]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).maximum_snow_albedo.tile5.nc, $(DATA)/]                                  
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).maximum_snow_albedo.tile6.nc, $(DATA)/]                                  
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).slope_type.tile1.nc, $(DATA)/]                                           
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).slope_type.tile2.nc, $(DATA)/]                                           
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).slope_type.tile3.nc, $(DATA)/]                                           
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).slope_type.tile4.nc, $(DATA)/]                                           
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).slope_type.tile5.nc, $(DATA)/]                                           
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).slope_type.tile6.nc, $(DATA)/]                                           
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).snowfree_albedo.tile1.nc, $(DATA)/]                                      
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).snowfree_albedo.tile2.nc, $(DATA)/]                                      
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).snowfree_albedo.tile3.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).snowfree_albedo.tile4.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).snowfree_albedo.tile5.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).snowfree_albedo.tile6.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).soil_type.tile1.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).soil_type.tile2.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).soil_type.tile3.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).soil_type.tile4.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).soil_type.tile5.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).soil_type.tile6.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).substrate_temperature.tile1.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).substrate_temperature.tile2.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).substrate_temperature.tile3.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).substrate_temperature.tile4.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).substrate_temperature.tile5.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).substrate_temperature.tile6.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).vegetation_greenness.tile1.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).vegetation_greenness.tile2.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).vegetation_greenness.tile3.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).vegetation_greenness.tile4.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).vegetation_greenness.tile5.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).vegetation_greenness.tile6.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).vegetation_type.tile1.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).vegetation_type.tile2.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).vegetation_type.tile3.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).vegetation_type.tile4.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).vegetation_type.tile5.nc, $(DATA)/]
parm/ufs/fix/gfs/land.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/sfc/$(CASE).mx$(OCNRES).vegetation_type.tile6.nc, $(DATA)/]
parm/ufs/fix/gfs/ocean.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/$(CASE).mx$(OCNRES)_oro_data.tile1.nc, $(DATA)/INPUT/oro_data.tile1.nc]
parm/ufs/fix/gfs/ocean.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/$(CASE).mx$(OCNRES)_oro_data.tile2.nc, $(DATA)/INPUT/oro_data.tile2.nc]
parm/ufs/fix/gfs/ocean.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/$(CASE).mx$(OCNRES)_oro_data.tile3.nc, $(DATA)/INPUT/oro_data.tile3.nc]
parm/ufs/fix/gfs/ocean.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/$(CASE).mx$(OCNRES)_oro_data.tile4.nc, $(DATA)/INPUT/oro_data.tile4.nc]                 [8/4604]
parm/ufs/fix/gfs/ocean.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/$(CASE).mx$(OCNRES)_oro_data.tile5.nc, $(DATA)/INPUT/oro_data.tile5.nc]
parm/ufs/fix/gfs/ocean.fixed_files.yaml:     - [$(FIXgfs)/orog/$(CASE)/$(CASE).mx$(OCNRES)_oro_data.tile6.nc, $(DATA)/INPUT/oro_data.tile6.nc]
ush/forecast_predet.sh:  FNALBC2=${FNALBC2:-"${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.facsf.tileX.nc"}
ush/forecast_predet.sh:  FNTG3C=${FNTG3C:-"${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.substrate_temperature.tileX.nc"}
ush/forecast_predet.sh:  FNVEGC=${FNVEGC:-"${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.vegetation_greenness.tileX.nc"}
ush/forecast_predet.sh:  FNVMNC=${FNVMNC:-"${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.vegetation_greenness.tileX.nc"}
ush/forecast_predet.sh:  FNVMXC=${FNVMXC:-"${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.vegetation_greenness.tileX.nc"}
ush/forecast_predet.sh:  FNSLPC=${FNSLPC:-"${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.slope_type.tileX.nc"}
ush/forecast_predet.sh:  FNALBC=${FNALBC:-"${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.snowfree_albedo.tileX.nc"}
ush/forecast_predet.sh:  FNVETC=${FNVETC:-"${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.vegetation_type.tileX.nc"}
ush/forecast_predet.sh:  FNSOTC=${FNSOTC:-"${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.soil_type.tileX.nc"}
ush/forecast_predet.sh:  FNSOCC=${FNSOCC:-"${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.soil_color.tileX.nc"}
ush/forecast_predet.sh:  FNABSC=${FNABSC:-"${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.maximum_snow_albedo.tileX.nc"}
ush/forecast_predet.sh:    ${NCP} "${FIXgfs}/orog/${CASE}/${CASE}_mosaic.nc" "${DATA}/INPUT/grid_spec.nc"
ush/forecast_predet.sh:    ${NCP} "${FIXgfs}/orog/${CASE}/${CASE}_mosaic.nc" "${DATA}/INPUT/${CASE}_mosaic.nc"
ush/forecast_predet.sh:    ${NCP} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile${tt}.nc" "${DATA}/INPUT/oro_data.tile${tt}.nc"
ush/forecast_predet.sh:    ${NCP} "${FIXgfs}/orog/${CASE}/${CASE}_grid.tile${tt}.nc"                 "${DATA}/INPUT/${CASE}_grid.tile${tt}.nc"
ush/gaussian_sfcanl.sh:#     fixed data : ${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile*.nc
ush/gaussian_sfcanl.sh:FIXWGTS=${FIXWGTS:-${FIXgfs}/orog/${CASE}/fv3_SCRIP_${CASE}_GRIDSPEC_lon${LONB_SFC}_lat${LATB_SFC}.gaussian.neareststod.nc}
ush/gaussian_sfcanl.sh:${NLN} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile1.nc" "./orog.tile1.nc"
ush/gaussian_sfcanl.sh:${NLN} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile2.nc" "./orog.tile2.nc"
ush/gaussian_sfcanl.sh:${NLN} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile3.nc" "./orog.tile3.nc"
ush/gaussian_sfcanl.sh:${NLN} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile4.nc" "./orog.tile4.nc"
ush/gaussian_sfcanl.sh:${NLN} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile5.nc" "./orog.tile5.nc"
ush/gaussian_sfcanl.sh:${NLN} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile6.nc" "./orog.tile6.nc"
ush/global_cycle.sh:#                   defaults to ${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.snowfree_albedo.tileX.nc
ush/global_cycle.sh:#                   Defaults to ${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.facsf.tileX.nc
ush/global_cycle.sh:#                   Defaults to ${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.substrate_temperature.tileX.nc
ush/global_cycle.sh:#                   Defaults to ${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.vegetation_greenness.tileX.nc
ush/global_cycle.sh:#                   Defaults to ${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.vegetation_type.tileX.nc
ush/global_cycle.sh:#                   Defaults to ${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.soil_type.tileX.nc
ush/global_cycle.sh:#                   Defaults to ${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.vegetation_greenness.tileX.nc
ush/global_cycle.sh:#                   Defaults to ${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.vegetation_greenness.tileX.nc
ush/global_cycle.sh:#                   Defaults to ${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.slope_type.tileX.nc
ush/global_cycle.sh:#                   Defaults to ${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.maximum_snow_albedo.tileX.nc
ush/global_cycle.sh:FNALBC2=${FNALBC2:-${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.facsf.tileX.nc}
ush/global_cycle.sh:FNTG3C=${FNTG3C:-${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.substrate_temperature.tileX.nc}
ush/global_cycle.sh:FNVEGC=${FNVEGC:-${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.vegetation_greenness.tileX.nc}
ush/global_cycle.sh:FNALBC=${FNALBC:-${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.snowfree_albedo.tileX.nc}
ush/global_cycle.sh:FNVETC=${FNVETC:-${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.vegetation_type.tileX.nc}
ush/global_cycle.sh:FNSOTC=${FNSOTC:-${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.soil_type.tileX.nc}
ush/global_cycle.sh:FNABSC=${FNABSC:-${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.maximum_snow_albedo.tileX.nc}
ush/global_cycle.sh:FNVMNC=${FNVMNC:-${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.vegetation_greenness.tileX.nc}
ush/global_cycle.sh:FNVMXC=${FNVMXC:-${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.vegetation_greenness.tileX.nc}
ush/global_cycle.sh:FNSLPC=${FNSLPC:-${FIXgfs}/orog/${CASE}/sfc/${CASE}.mx${OCNRES}.slope_type.tileX.nc}
ush/global_cycle_driver.sh:  ln -fs $FIXgfs/orog/${CASE}/C${CRES}_grid.tile${n}.nc       $DATA/fngrid.00$n
ush/global_cycle_driver.sh:    ln -fs $FIXgfs/orog/${CASE}/C${CRES}_oro_data.tile${n}.nc   $DATA/fnorog.00$n
ush/global_cycle_driver.sh:    ln -fs $FIXgfs/orog/${CASE}/C${CRES}.mx${OCNRES}_oro_data.tile${n}.nc   $DATA/fnorog.00$n
```

WalterKolczynski-NOAA commented:
@guoqing-noaa I think changing it is still the way to go. Changing UFS Utils may require additional changes beyond scripts we use. @GeorgeGayno-NOAA will know for sure

@GeorgeGayno-NOAA Short version: We're looking to go back to using `${FIXorog}` in place of `${FIXgfs}/orog` as the best way to support running with a nest for the AR project, which needs a different set of orog and ugwd fix files. I know we only recently changed it, so apologies.

GeorgeGayno-NOAA commented:
> @guoqing-noaa I think changing it is still the way to go. Changing UFS Utils may require additional changes beyond scripts we use. @GeorgeGayno-NOAA will know for sure
> 
> @GeorgeGayno-NOAA Short version: We're looking to go back to using `${FIXorog}` in place of `${FIXgfs}/orog` as the best way to support running with a nest for the AR project, which needs a different set of orog and ugwd fix files. I know we only recently changed it, so apologies.

That change was done in this PR: https://github.com/ufs-community/UFS_UTILS/pull/899. Does the entire PR need to be reverted or just the `$FIXorog` part?

guoqing-noaa commented:
Thanks, @WalterKolczynski-NOAA and @GeorgeGayno-NOAA 

Currently, the idea is to revert `$FIXorog` and `$FIXugwd` only

I will work on this.

GeorgeGayno-NOAA commented:
> Thanks, @WalterKolczynski-NOAA and @GeorgeGayno-NOAA
> 
> Currently, the idea is to revert `$FIXorog` and `$FIXugwd` only
> 
> I will work on this.

I will open a UFS_UTILS issue? Should I assign it to you or myself?

guoqing-noaa commented:
> I will open a UFS_UTILS issue? Should I assign it to you or myself?

Thanks, @GeorgeGayno-NOAA!  It will be great if you can make changes for UFS_UTILS. And I can focus on making changes to the global workflow.

aerorahul commented:
@guoqing-noaa @GeorgeGayno-NOAA 
I agree w/ @WalterKolczynski-NOAA.
We should define `FIXorog` and `FIXugwd` depending on `DO_NEST` where applicable and use those 2 variables instead of `$FIXgfs/orog` and `$FIXgfs/ugwd`.  The variables `FIXorog` and `FIXugwd` could be defined in `config` files or in the j-job that needs this information.
@GeorgeGayno-NOAA 
Someone from the workflow team can make that change in the ufs-utils.
@guoqing-noaa 
If you can make the update for the nest in the global-workflow in this PR, we could update the hash of ufs-utils included in this PR.
Please let me know if we need to coordinate/clarify offline.

GeorgeGayno-NOAA commented:
> @guoqing-noaa @GeorgeGayno-NOAA I agree w/ @WalterKolczynski-NOAA. We should define `FIXorog` and `FIXugwd` depending on `DO_NEST` where applicable and use those 2 variables instead of `$FIXgfs/orog` and `$FIXgfs/ugwd`. The variables `FIXorog` and `FIXugwd` could be defined in `config` files or in the j-job that needs this information. @GeorgeGayno-NOAA Someone from the workflow team can make that change in the ufs-utils. 

Great. I opened this issue: https://github.com/ufs-community/UFS_UTILS/issues/955

>@guoqing-noaa If you can make the update for the nest in the global-workflow in this PR, we could update the hash of ufs-utils included in this PR. Please let me know if we need to coordinate/clarify offline.

guoqing-noaa committed:
Merge branch 'develop' into pr_nest_fix_files

guoqing-noaa commented:
Thanks, @aerorahul. I will update this PR sometime today and get ready for a review.

aerorahul commented:
@guoqing-noaa 
Please see https://github.com/ufs-community/UFS_UTILS/pull/956
You should be able to define and export `FIXorog` in the config files or in the script where `global_cycle.sh` is called to realize the change.

guoqing-noaa committed:
Changes based on reviews
- link nest fix files when needed, i.e. keep the --nest option
- change $FIXgfs/orog to $FIXorog
- change $FIXgfs/ugwd to $FIXugwd

guoqing-noaa committed:
update .gitignore

guoqing-noaa commented:
> @guoqing-noaa Please see [ufs-community/UFS_UTILS#956](https://github.com/ufs-community/UFS_UTILS/pull/956) You should be able to define and export `FIXorog` in the config files or in the script where `global_cycle.sh` is called to realize the change.
Thanks! @aerorahul

guoqing-noaa committed:
Merge branch 'develop' into pr_nest_fix_files

guoqing-noaa commented:
@aerorahul @WalterKolczynski-NOAA This PR is ready for review. Let me know if you have any comments. Thanks!

WalterKolczynski-NOAA reviewed (changes requested):
Looks good except the UFS_UTILS version needs to be updated. Let us know if you need guidance on how to update a submodule hash.

guoqing-noaa committed:
Merge branch 'develop' into pr_nest_fix_files

guoqing-noaa committed:
update ufs_utils hash

guoqing-noaa commented:
> Looks good except the UFS_UTILS version needs to be updated. Let us know if you need guidance on how to update a submodule hash.

@WalterKolczynski-NOAA Thanks for the feedback. I just updated the UFS_UTILS hash.

WalterKolczynski-NOAA reviewed (approved):
Conditionally approved pending successful completion of CI tests.

WalterKolczynski-NOAA reviewed (changes requested):
`parm/config/gefs/config.base` needs to be updated to define `FIXorog` and `FIXugwd`

guoqing-noaa committed:
add FIXorog and FIXugwd in parm/config/gefs/config.base

guoqing-noaa commented:
> `parm/config/gefs/config.base` needs to be updated to define `FIXorog` and `FIXugwd`

updated. Thanks, @WalterKolczynski-NOAA
